mozmeao/birdbox
# SSO support

This changeset adds SSO authentication via Mozilla OIDC and Auth0 so that, in a deployed site (or if configured locally to use SSO), a user _has_ to use the Mozilla SSO flow to sign in. 

Plus, only users who are already added to the Wagtail admin will be able to authenticate - we aren't allowing 'drive by' signups to the Wagtail admin - it'll be needed-access only.

~Of course, for the first user in a system, we need a way to easily set up _their_ account, so we try to solve that by auto-provisioning admin accounts based on an env var, which would be injected into the environment from a stored GCP Secret.~ This has been dropped in favour of manual setup of that first user, which will be documented internally


## To test

_You can do this with local build or Docker, if you use docker, the env file to edit is `/path/to/docker/envfiles/local.env`; if you use local, it's a `.env` file in the root of the project_

* Check out this branch
* Set up an env file with the development OIDC credentials passed to you by @stevejalim
* In the env file, add var called `BIRDBOX_ADMIN_USER_EMAILS` with the evalue of "sjalim@mozilla.com,YOUREMAIL@mozilla.com" replacing YOUREMAIL with, well, your email
* You should not need to set an env var of `USE_SSO_AUTH=True` because that's the default value
* Start birdbox running (`just run-local` or `docker-compose up app --build`)
* Go to /admin/ and you should see a button to login with SSO. Click it and you should go through the OAuth flow and end up in the Wagtail admin wiht an account auto-created for you (because you're an admin).
* Follow the menu from Settings > Users and confirm there's also a sjalim@mozilla.com admin created along with yourself.
* Sign out from bottom right
* Go to /django-admin/ and confirm you can sign in there with SSO, too

### Example screenshots

<img width="793" alt="Screenshot 2023-09-13 at 21 37 51" src="https://github.com/mozmeao/birdbox/assets/101457/3e27d08f-11ec-49f5-9d05-ce72849e8c8a">


<img width="665" alt="Screenshot 2023-09-13 at 14 21 09" src="https://github.com/mozmeao/birdbox/assets/101457/be3bc7d3-a4a0-435b-b895-41fd33471c02">


Also, here's an example of the `SuspiciousOperation` being raised should someone manage to get a non-Mozilla email into the env var for bootstrapping admins

<img width="781" alt="Screenshot 2023-09-13 at 22 36 05" src="https://github.com/mozmeao/birdbox/assets/101457/f5a33732-fa95-4106-b314-e27f06319984">


----

Resolves https://github.com/mozmeao/birdbox/issues/45
